docs/source/developer/coding_standards.rst

```diff
@@ -0,0 +1,125 @@
+* If applicable, the pull request should include updates to the :ref:`documentation <documentation>`.
+* All pull requests should include any information required to test the changes. This may include:
+
+  * A description of the changes
+  * A description of how to test the changes
+  * Any relevant data or scripts required to test the changes
+
+A simple example of this workflow would be as follows:
+
+.. code-block:: sh
+
+    git checkout -b my_new_feature
+    # Write Tests for your new feature
+    python -m pytest # New tests should fail
+    # Write code to make the tests pass
+    python -m pytest # All tests should pass
+    # update the documentation (if applicable)
+    git add <files>
+    git commit -m "My new feature" # pre-commit will run here
+    git push -u origin my_new_feature
+
+Then, go to the `github page <https://github.com/neutrons/garnet/>`_ for the repository and create a new pull request from the ``my_new_feature`` branch to the ``next`` branch.
+
+================
+Coding Standards
+================
+
+A few notes on coding standards:
+
+* Variable and function names should be descriptive and follow the ``lower_case_with_underscores`` convention (commonly referred to as ``snake_case``).
+* Filenames should be lowercase and use underscores to separate words.
+* All code will be checked and formatted using `ruff <https://docs.astral.sh/ruff/rules/>`_.
+* Docstrings should be used to document all functions, classes, and Modules.
+* Pre-commit hooks are configured to run for every PR, and can be run locally on every commit.
+
+Pre-commit
+----------
+
+Pre-commit is a tool that is used to ensure that all code is formatted and linted before being committed.
+
+Install pre-commit
+++++++++++++++++++
+
+pre-commit can be installed using the following command:
+
+.. code-block:: sh
+
+    pre-commit install
+
+You should only need to do this once. pre-commit will now run on every commit.
+
+Run pre-commit manually
++++++++++++++++++++++++
+
+The following command will run pre-commit on all staged changes:
+
+.. code-block:: sh
+
+    pre-commit
+
+To run pre-commit on all files in the repository, run the following command:
+
+.. code-block:: sh
+
+    pre-commit run --all
+
+More information on pre-commit can be found in the `pre-commit documentation <https://pre-commit.com/>`_.
+
+
+========================
+Documentation Guidelines
+========================
+
+.. _documentation:
+
+Updating the Documentation
+--------------------------
+This project uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_ to generate documentation and is updated with every pull request.
+The documentation is a combination of pages written in `reStructuredText <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_ and
+docstrings pulled from the code.
+A simple guide to updating documentation in Garnet can be found `here <https://github.com/neutrons/garnet/blob/next/docs/README.md>`_.
```

Comment:
did you mean to put a link to the repository instead of a sphinx page?

Reply:
Yeah I can make it a sphinx page if you would like but that is more of a "meta" page for developers to update documentation not something that I felt should be published but maybe I either need to not link to the page or make it a sphinx page?